Source/SwiftLintFramework/Extensions/String+SwiftLint.swift

```diff
@@ -69,4 +69,16 @@ extension String {
         }
         return nil
     }
+
+    var stringByDeletingLastPathComponent: String {
```

Comment:
For methods on NSString, I prefer not masking them behind a `Swift.String` interface, since that hides the fact that the string has to be cast to NSString. Especially since SwiftLint (and SourceKitten) make heavy use of NSString in its implementation, you then end up casting an NSString into a `Swift.String` to use these methods, which then cast it back to an NSString.

For these reasons, I prefer casting at the call site.